packages/forms/src/lib/upload/README.md

```diff
@@ -0,0 +1,276 @@
+| `@Output() invalidFiles: EventEmitter<InvalidFile[]>;` | - | Emits an array of invalid files and the reason why they are invalid. |
+
+#### `<aui-upload-queue>`
+This component provides a list of queued files.
+
+| Name         | Default value | Description |
+| -----------  | ------ | -------------------------- |
+| `@Input() files: File[];` | - | An array of files. |
+| `@Input() uploader: Uploader;` | - | An instance of the Uploader class. This class handles the validation and upload of a file. Its different options and there default values are explained in detail above. |
+| `@Output() uploadedFiles: EventEmitter<Object[]>` | - | Emits an array of files that were uploaded. |
+
+#### `<aui-validation-list>`
+This component displays possible error messages.
+
+| Name         | Default value | Description |
+| -----------  | ------ | -------------------------- |
+| `@Input() invalidFiles: InvalidFile[];` | `[]` | An array of invalid files and the reason why they are invalid. |
+
+#### `Uploader`
+This class handles the validation and upload of a file.
+
+##### `UploadOptions`
+
+| Name         | Default value | Description |
+| -----------  | ------ | -------------------------- |
+| `allowedMimeTypes: string[];` | `[]` | The MIME types that are allowed to be uploaded. |
+| `allowedFileTypes: string[];` | `[]` | The file extensions that are allowed to be uploaded. |
+| `autoUpload: boolean;` | `false` | Upload the file automatically without confirmation. |
+| `maxFileSize: number;` | `0` | The maximum file size that is allowed. The value is expressed in Bytes. The value 0 equals infinite. |
+| `queueLimit: number;` | `0` | The maximum number of files that can be selected at the same time. The value 0 equals infinite. |
+| `type: string;` | `drop` | By default an upload has an `upload-zone` but with the option `button`, you can set it as an uploadbutton. |
+| `url: string;` | `''` | The upload url. |
+
+#### Example
+
+```typescript
+import { UploadModule } from '@acpaas-ui/ngx-components/forms';
+
+@NgModule({
+    imports: [
+        UploadModule
+    ]
+});
+
+export class AppModule {};
+```
+
+**Basic upload**
+
+```typescript
+public dropzone1: UploadOptions = {
+    allowedMimeTypes: ['image/jpeg'],
+    maxFileSize: 10000000,
+    queueLimit: 2,
+    type: 'drop',
+    url: 'api/upload',
+};
+```
+
+```html
+<aui-upload [options]="dropzone1" (selectUploadedFiles)="onUpload($event)">
+    <div class="aui-upload-message">
+        Drag your files here or click to upload
+    </div>
+    <div class="aui-upload-description">
+        Optional description message
+    </div>
+</aui-upload>
+```
+
+**Upload button**
+
+```typescript
+public dropzone2: UploadOptions = {
+    type: 'button',
+    allowedFileTypes: ['.jpg', 'jpeg', 'png'],
+    autoUpload: true,
+    maxFileSize: 2000000,
+    url: 'api/upload',
+};
+```
+
+```html
+<div class="u-margin-bottom">
```

Comment:
I guess there's something wrong with your code editor. It doesn't seem to complain about the tabs (that should be spaces).